File: My_Data_Structure_with_python/Stack/Stack.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class Stack:
    def __init__(self):
        self.head = None

    def is_empty(self):
        return self.head is None

    def push(self, data):
        new_node = Node(data)
        new_node.next = self.head
        self.head = new_node

    def pop(self):
        if self.is_empty():  # self.head is None:
            raise ValueError("Stack is Empty!")
        popped_node = self.head
        self.head = self.head.next
        popped_node.next = None
        return popped_node.data

    def peek(self):
        if self.is_empty(): # if self.head is None:
           raise ValueError("Stack is Empty!")
        return self.head.data

    def display(self):
        if self.is_empty():
            print("Stack is empty")
        else:
            current = self.head
            while current:
                print(current.data, end=" ")
                current = current.next
            print()

# Example usage:
stack = Stack()
stack.push(1)
stack.push(2)
stack.push(3)
stack.display()  # Output: 3 2 1
print(stack.peek())  # Output: 3
print(stack.pop())  # Output: 3
stack.display()  # Output: 2 1
print(stack.is_empty())

File: Data_Structure_java/Linked List/Stack/Stack.java
import java.util.EmptyStackException;

class Node
{
    int data;
    Node link;
    Node(int data)
    {
        this.data=data;
        this.link=null;
    }
}
class Stack
{
    Node head;
    Stack()
    {
        this.head=null;
    }
    public boolean isEmpty()
    {
        return this.head==null;
    }
    public void push(int data)
    {
        Node newNode=new Node(data);
        newNode.link=this.head;
        this.head=newNode;
    }
    public int pop()
    {
        if (isEmpty())
            throw new EmptyStackException();

        int poped_data=this.head.data;
        this.head=this.head.link;
        return poped_data;
    }
    public void display() {
        if (isEmpty()) {
            System.out.println("Stack is empty.");
            return;
        }

        Node current = this.head;
        System.out.print("Stack (top to bottom): ");
        while (current != null) {
            System.out.print(current.data + " ");
            current = current.link;
        }
        System.out.println();
    }
}
class Stack1
{
    public static void main(String[] args) 
    {
        Stack head=new Stack();
        head.push(10);
        head.push(14);
        head.push(60);
        head.push(100);

        int n=head.pop();
        head.display();

    }
}